### DISCUSSION TITLE
Loader function is not executed, when optional params are not set. Bug?

### DISCUSSION THREAD

**Discussion by itshallrun** [TanStack/router#1839] [Q&A]
I could narrow the issue down, and [here](https://stackblitz.com/edit/tanstack-router-sazibb?file=src%2Froutes%2F__root.tsx) is a basic reproduction of it.

When navigation from anything else then tanstack router without setting the optional search params in the URL to at least an empty value, the loader function is not run, and the data is not fetched.
The consecutive Components which rely on the loader data, obviously fail then.

The example is very basic. It has two links. A tanstack link, and a non tanstack link.
Click on both, and you will see that the non Tanstack link, does not execute the loader.
![image](https://github.com/TanStack/router/assets/21173275/6d76d146-ceb9-45ac-8428-50043c70057c)

I am using a non tanstack link here, just to visualize, that there is a problem. 

The problem also arises, when reloading a page, where the optional search params are missing:

For more examples, here are some videos I recorded with the [basic-default-search-params](https://github.com/TanStack/router/tree/main/examples/react/basic-default-search-params) example

https://github.com/TanStack/router/assets/21173275/bb03abc6-df2d-4a1f-9f08-e15499eda330

https://github.com/TanStack/router/assets/21173275/69a28baa-cc1a-44e3-801d-8b31eee9f6e2

As you can see, reloading the page, or redirecting with anything other than tanstack link, leads to an error.
In the video I used `window.location.href` for the demonstration

This is fine
`window.location.href = "http://localhost:5174/posts/post?postId=2&color=red&paramWithoutDefault="
`
Leads to an error
`window.location.href = "http://localhost:5174/posts/post?postId=2&color=red"
`

The error is thrown, because the Loader function is not executed:

This is how it should be, and is when the params are all set:
![image](https://github.com/TanStack/router/assets/21173275/7267f5d6-4ca8-4001-b982-565f738191fa)

This is how it looks, when I reload the page and one of the optional params is not set:

![image](https://github.com/TanStack/router/assets/21173275/d5c5fc4c-3c7a-4b25-a293-56f123cbfcb2)

It seems like there is a bug with the loader function

**Comment by itshallrun** [comment]
@SeanCassiere could you please take a brief look at this? Does it deserve a bug report, or is it a misunderstanding on my side?

**Comment by schiller-manuel** [comment]
this will be fixed by #1907 

here is your reproducer with the pre-release package: https://stackblitz.com/edit/tanstack-router-cd6bqy?file=src%2Froutes%2F__root.tsx
